File: src/com/dmgburg/visualization/PolicyNode.java
package com.dmgburg.visualization;

import java.util.ArrayList;
import java.util.List;

public class PolicyNode {
    public List<PolicyNode> children = new ArrayList<PolicyNode>();
    public Desicion decision;
    public String description;
    public String id;

    public PolicyNode(){
        description = "";
    }

    public PolicyNode(String description, Desicion decision){
        this(description);
        this.decision = decision;
    }

    public PolicyNode addChild(PolicyNode child){
        children.add(child);
        return this;
    }

    public PolicyNode(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "PolicyNode{" +
                "children=" + children +
                ", decision=" + decision +
                ", description='" + description + '\'' +
                ", id='" + id + '\'' +
                '}';
    }
}

File: src/com/dmgburg/visualization/PolicyNode.kt
package com.dmgburg.visualization

import com.fasterxml.jackson.annotation.JsonIgnore
import java.util.ArrayList

class PolicyNode {
    @JsonIgnore
    var parent: PolicyNode? = null
    var children: MutableList<PolicyNode> = ArrayList()
    var decision: Desicion? = null
    var description: String
    var id: String? = null

    constructor(parent: PolicyNode?) {
        this.parent = parent
        description = ""
    }

    constructor(parent: PolicyNode?, description: String) {
        this.description = description
    }

    constructor(parent: PolicyNode,description: String, decision: Desicion) : this(parent, description) {
        this.decision = decision
    }

    fun addChild(child: PolicyNode): PolicyNode {
        children.add(child)
        return this
    }

    override fun toString(): String {
        return "PolicyNode{id='" + id + '\'' +'}'
    }
}
